### PR TITLE
Jibri rotation

### PR DESCRIPTION
**Pull request by oanaianc**

### CODE REVIEW

File: src/oracle_instance_manager.ts [hunk 1]
```diff
@@ -137,6 +143,20 @@ export default class OracleInstanceManager {
                 metadata: { group: groupName },
             };
             await this.jibriTracker.track(ctx, state);
+            if (isScaleDownProtected) {
+                await this.shutdownManager.setScaleDownProtected(
+                    ctx,
+                    {
+                        instanceId: launchResponse.instance.id,
+                        instanceType: 'jibri',
```
Comment (aaronkvanmeerten) [inline src/oracle_instance_manager.ts:151]: Is this safe to hardcode as 'jibri' ?  Shouldn't this be set from to the type property on the group?

File: src/shutdown_manager.ts [hunk 2]
```diff
@@ -31,9 +35,25 @@ export default class ShutdownManager {
...
-        if (res == 'shutdown') {
-            return true;
-        }
-        return false;
+        return res == 'shutdown';
+    }
+
+    async setScaleDownProtected(
+        ctx: Context,
+        details: InstanceDetails,
```
Comment (aaronkvanmeerten) [inline src/shutdown_manager.ts:43]: Do we need the details here?  We seem to only use the instanceId, so perhaps it makes more sense to only pass the instanceId?

File: src/shutdown_manager.ts [hunk 3]
```diff
@@ -31,9 +35,25 @@ export default class ShutdownManager {
...
+        protectedTTL: number,
+        mode = 'isScaleDownProtected',
+    ): Promise<boolean> {
+        const key = this.protectedKey(details.instanceId);
+        ctx.logger.debug('Writing protected mode', { key, mode });
+        await this.redisClient.set(key, mode, 'ex', protectedTTL);
+        return true;
+    }
+
+    async isScaleDownProtected(ctx: Context, details: InstanceDetails): Promise<boolean> {
```
Comment (aaronkvanmeerten) [inline src/shutdown_manager.ts:53]: Same question here: do we need the full instance details here?  Seems like if we're only using instanceId, then passing only that value would work and not require knowing the other instance details?  I'm open to discussing, don't feel strongly about it.

File: src/handlers.ts [hunk 4]
```diff
@@ -9,6 +9,9 @@ import ShutdownManager from './shutdown_manager';
 interface SidecarResponse {
     shutdown: boolean;
 }
+interface JibriDesired {
```
Comment (aaronkvanmeerten) [inline src/handlers.ts:12]: Should this be called "Jibri" ?

Comment (oanaianc) [reply src/handlers.ts:12]: No, I'm renaming it

**Review by raluca8x8 — APPROVED** [review]

**Pull request merged**